syntax = "proto3";

package auth;

import "google/api/annotations.proto";


service AuthService {
  rpc Login (LoginRequest) returns (LoginResponse) {
    option (google.api.http) = {
      post: "/v1/login",
      body: "*"
    };
  }
  rpc Auth (AuthRequest) returns (AuthResponse) {
    option (google.api.http) = {
      post: "/v1/auth",
      body: "*"
    };
  }
}

message LoginRequest {
  string userLogin = 1;
}
message LoginResponse {
  string accessToken = 1;
}

message AuthRequest {
  string accessToken = 1;
}
message AuthResponse {
  string userLogin = 1;
  bool valid = 2;
}
